addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3066,8 +3066,14 @@ options.registry.WebsiteAnimate = options.Class.extend({
      */
     async start() {
         await this._super(...arguments);
+        const window = this.ownerDocument.defaultView;
+        this.$window = window.$(window);
+        // Animations for which the "On Scroll" and "Direction" options are not
+        // available.
+        this.limitedAnimations = ['o_anim_flash', 'o_anim_pulse', 'o_anim_shake', 'o_anim_tada', 'o_anim_flip_in_x', 'o_anim_flip_in_y'];
         this.isAnimatedText = this.$target.hasClass('o_animated_text');
         this.$optionsSection = this.$overlay.data('$optionsSection');
+        this.$scrollingElement = $().getScrollingElement(this.ownerDocument);
```

Comment:
It works because of stuff like that :
```js
            const scrollTop = document.body.classList.contains('modal-open') ?
                this.$target.find('.modal:visible').scrollTop() :
                this.$scrollingElement.scrollTop();
```
And also thanks to the `capture: true` on the `EventListener` I think

Reply:
Yes ok then my question is more: why do you need to initialize `$scrollingElement` to the page scrolling element in case the `$target` is in a modal?